python3.pkgs.ipykernel 5.2.1 build failure on darwin

**Describe the bug**
Ipykernel 5.2.1 (via #87223) introduced a test for UNC paths on windows which fail on darwin.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build -A python3.pkgs.ipykernel`

**Expected behavior**
Build of ipykernel for python 3.7 to succeed.

**Logs**

```
============================= test session starts ==============================
platform darwin -- Python 3.7.7, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /private/tmp/nix-build-python3.7-ipykernel-5.2.1.drv-0/ipykernel-5.2.1
plugins: flaky-3.6.1
collected 108 items / 16 deselected / 92 selected

ipykernel/inprocess/tests/test_kernel.py .s..                            [  4%]
ipykernel/inprocess/tests/test_kernelmanager.py .....                    [  9%]
ipykernel/tests/test_async.py ..ss                                       [ 14%]
ipykernel/tests/test_connect.py .....                                    [ 19%]
ipykernel/tests/test_embed_kernel.py ...                                 [ 22%]
ipykernel/tests/test_eventloop.py .                                      [ 23%]
ipykernel/tests/test_heartbeat.py ....                                   [ 28%]
ipykernel/tests/test_io.py .                                             [ 29%]
ipykernel/tests/test_jsonutil.py .....                                   [ 34%]
ipykernel/tests/test_kernel.py ....s...s.F                               [ 46%]
ipykernel/tests/test_kernelspec.py ........                              [ 55%]
ipykernel/tests/test_message_spec.py ......................              [ 79%]
ipykernel/tests/test_pickleutil.py .....                                 [ 84%]
ipykernel/tests/test_serialize.py ....ssss......                         [100%]

=================================== FAILURES ===================================
________________________________ test_unc_paths ________________________________

args = (), kwargs = {}

    def skipper_func(*args, **kwargs):
        """Skipper for normal test functions."""
        if skip_val():
            raise nose.SkipTest(get_msg(f,msg))
        else:
>           return f(*args, **kwargs)

/nix/store/2ysz669wb8qbn3q4cybkgrzjw9czp6gk-python3.7-ipython-7.13.0/lib/python3.7/site-packages/IPython/testing/decorators.py:212:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    @dec.skipif(sys.platform.startswith('linux'))
    def test_unc_paths():
        with kernel() as kc, TemporaryDirectory() as td:
            drive_file_path = os.path.join(td, 'unc.txt')
            with open(drive_file_path, 'w+') as f:
                f.write('# UNC test')
            unc_root = '\\\\localhost\\C$'
            file_path = os.path.splitdrive(os.path.dirname(drive_file_path))[1]
            unc_file_path = os.path.join(unc_root, file_path[1:])

            iopub = kc.iopub_channel

            kc.execute("cd {0:s}".format(unc_file_path))
            reply = kc.get_shell_msg(block=True, timeout=TIMEOUT)
            assert reply['content']['status'] == 'ok'
            out, err = assemble_output(iopub)
>           assert unc_file_path in out
E           assert '\\\\localhost\\C$/private/tmp/nix-build-python3.7-ipykernel-5.2.1.drv-0/tmpa1ace7xo' in "[Errno 2] No such file or directory: '\\\\localhostC$/private/tmp/nix-build-python3.7-ipykernel-5.2.1.drv-0/tmpa1ace7xo'\n/private/tmp/nix-build-python3.7-ipykernel-5.2.1.drv-0/ipykernel-5.2.1\n"

ipykernel/tests/test_kernel.py:337: AssertionError
```

**Additional context**
Adding the test to `disabledTests` solves this

**Notify maintainers**
@FRidh 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.6.0, macOS 10.14.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixpkgs-20.03pre207249.7e8454fb856"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/python-modules/ipykernel/default.nix
+++ b/pkgs/development/python-modules/ipykernel/default.nix
@@ -43,7 +43,7 @@ buildPythonPackage rec {
     "test_subprocess_error"
     "test_ipython_start_kernel_no_userns"
     
-    # see https://github.com/NixOS/nixpkgs/issues/87626
+    # https://github.com/ipython/ipykernel/issues/506
     "test_unc_paths"    
   ] ++ lib.optionals (pythonOlder "3.8") [
     # flaky test https://github.com/ipython/ipykernel/issues/485

--- a/pkgs/development/python-modules/ipykernel/default.nix
+++ b/pkgs/development/python-modules/ipykernel/default.nix
@@ -42,6 +42,9 @@ buildPythonPackage rec {
     "test_subprocess_print"
     "test_subprocess_error"
     "test_ipython_start_kernel_no_userns"
+    
+    # see https://github.com/NixOS/nixpkgs/issues/87626
+    "test_unc_paths"    
   ] ++ lib.optionals (pythonOlder "3.8") [
     # flaky test https://github.com/ipython/ipykernel/issues/485
     "test_shutdown"

